client/asset/btc/btc.go

```diff
@@ -3928,6 +3987,79 @@ func preAccelerate(btc *baseWallet, swapCoins, accelerationCoins []dex.Bytes, ch
 	return currentEffectiveRate, &suggestedRange, earlyAcceleration, nil
 }
 
+func (btc *baseWallet) txDB() txDB {
+	dbi := btc.txHistoryDB.Load()
+	if dbi == nil {
+		return nil
+	}
+	return dbi.(txDB)
+}
+
+func (btc *baseWallet) markTxAsSubmitted(id dex.Bytes) {
+	txHistoryDB := btc.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
+	var hash chainhash.Hash
+	copy(hash[:], id)
+
+	btc.pendingTxsMtx.Lock()
+	wt, found := btc.pendingTxs[hash]
+	if found {
+		wt.mtx.Lock()
+		wt.Submitted = true
+		wt.mtx.Unlock()
+	}
+	btc.pendingTxsMtx.Unlock()
+
+	err := txHistoryDB.markTxAsSubmitted(id)
+	if err != nil {
+		btc.log.Errorf("failed to mark tx as submitted in tx history db: %v", err)
+	}
+}
+
+func (btc *baseWallet) removeTxFromHistory(id dex.Bytes) {
+	txHistoryDB := btc.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
+	err := txHistoryDB.removeTx(id)
+	if err != nil {
+		btc.log.Errorf("failed to remove tx from tx history db: %v", err)
+	}
+}
+
+func (btc *baseWallet) addTxToHistory(txType asset.TransactionType, id dex.Bytes, balanceDelta int64, fees uint64, submitted bool) {
+	txHistoryDB := btc.txDB()
+	if txHistoryDB == nil {
+		return
+	}
+
```

Comment:
Don't we want to add the tx to the `pendingTxs` map?

Reply:
No, this isn't like ETH where we're using the `pendingTxs` to calculate a pending balance. `pendingTxs` is just now getting added in this PR. It's just used to keep track of which transactions in the db may need to be updated whenever there's a new block.